pyexchange/tethfinex.py

```diff
@@ -211,50 +215,105 @@ def get_orders(self, pair: str) -> List[Order]:
+                else:
+                    return float(price_string[:index + precision])
+
     def place_order(self,
                     is_sell: bool,
                     pay_token: Address,
                     pay_amount: Wad,
                     buy_token: Address,
                     buy_amount: Wad,
-                    fee_address: Address,
+                    ethfinex_address: Address,
                     pair: str) -> Order:
 
         assert(isinstance(is_sell, bool))
         assert(isinstance(pay_token, Address))
         assert(isinstance(pay_amount, Wad))
         assert(isinstance(buy_token, Address))
         assert(isinstance(buy_amount, Wad))
-        assert(isinstance(fee_address, Address))
+        assert(isinstance(ethfinex_address, Address))
         assert(isinstance(pair, str))
 
+        # check if symbol pair exists in tethefinex exchange
+        symbols_details = self.get_symbols_details()
+        pair_exist = False
+        precision = 0
+        for symbol in symbols_details:
+            if symbol['pair'] == pair.lower():
+                pair_exist = True
+                precision = symbol["price_precision"]
+        assert(pair_exist)
+
+        symbol_one = pair[0:3]
+        symbol_two = pair[3:6]
+
+        sell_symbol = symbol_one if is_sell else symbol_two
+        sell_amount = pay_amount if is_sell else buy_amount
+        sell_token = pay_token if is_sell else buy_token
+
+        buy_symbol = symbol_two if is_sell else symbol_one
+        buy_amount = buy_amount if is_sell else pay_amount
+        buy_token = buy_token if is_sell else pay_token
+
+        # retrieve the minimun order size for each pair symbol
+        ethfinex_config = self.get_config()['0x']
+        sell_currency_min_order = Wad.from_number(ethfinex_config['tokenRegistry'][sell_symbol]['minOrderSize'])
+        buy_currency_min_order =  Wad.from_number(ethfinex_config['tokenRegistry'][buy_symbol]['minOrderSize'])
+
+        # fee rate
+        # use 0.0025 for 0.25% fee
+        # use 0.01   for 1% fee
+        # use 0.05   for 5% fee
+        fee_rate = Wad.from_number(0.0025)
+        price = float(buy_amount / sell_amount) if is_sell else float(sell_amount / buy_amount)
+        # return price with precision
+        price = self.add_price_precision(price, precision)
+        amount = round(float(sell_amount),6) if is_sell else round(float(buy_amount),6)
+
+        buy_amount_order_no_fees = Wad.from_number(price * amount) if is_sell else buy_amount
+        buy_amount_order = buy_amount_order_no_fees - buy_amount_order_no_fees/Wad.from_number(100) * fee_rate * Wad.from_number(100)
+        buy_asset_order = Asset.deserialize('0xf47261b0E' + str(buy_token)[2:])
+        sell_asset_order = Asset.deserialize('0xf47261b0E' + str(sell_token)[2:])
+        sell_amount_order = Wad.from_number(amount) if is_sell else Wad.from_number(price * amount)
+        assert(buy_amount_order_no_fees >= buy_currency_min_order)
```

Comment:
I suggest not to assert here but leave call to API throw error and calling party (that is keeper) handle it